libraries/Microsoft.Bot.Builder/Inspection/InspectionMiddleware.cs

```diff
@@ -0,0 +1,198 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Bot.Connector.Authentication;
+using Microsoft.Bot.Schema;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Bot.Builder
+{
+    public class InspectionMiddleware : InterceptionMiddleware
+    {
+        private InspectionState _inspectionState;
+        private UserState _userState;
+        private ConversationState _conversationState;
+        private MicrosoftAppCredentials _credentials;
+
+        public InspectionMiddleware(InspectionState inspectionState, UserState userState = null, ConversationState conversationState = null, MicrosoftAppCredentials credentials = null, ILogger<InspectionMiddleware> logger = null)
+            : base(logger)
+        {
+            _inspectionState = inspectionState;
+            _userState = userState;
+            _conversationState = conversationState;
+            _credentials = credentials ?? MicrosoftAppCredentials.Empty;
+            HttpClient = new HttpClient();
+        }
```

Comment:
This makes me kinda queasy. Is there a way to avoid this?

Reply:
done